src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -341,6 +341,7 @@ void distributeFlushTasks() {
       List<List<ChannelData<T>>> blobData = new ArrayList<>();
       float totalBufferSize = 0;
 
+      final String filePath = getFilePath(this.targetStage.getClientPrefix());
```

Comment:
The issue with moving this here is that the blob name sequencer would have gaps now, because it's possible that blobData is empty, and we kind of replying on the continuous sequencer in the blob name to figure out the name of the next blob during debugging, could you take a look at how it was done in https://github.com/snowflakedb/snowflake-ingest-java/pull/136/files and update it accordingly? Basically we want to save the previous filePath if the blobData is empty

Reply:
I see. Thanks for the explanation @sfc-gh-tzhang . I will integrate this.

Reply:
I checked the code in the PR you mentioned and I think that it does not prevent gaps totally. Imagine that we have 5 channels, we create a filename for the 5th (so we bump the counter) and that channel is empty so we do nothing and we exit the method. Now the `filePath` variable you have in the PR that keeps your previous filename is deleted because we exited the method. At the next iteration, we will re-enter this method and bump again the counter, so now we will have gaps, right?